src/interactive/repl.ts

```diff
@@ -0,0 +1,414 @@
+        if (rx.test(doc.lineAt(start).text)) {
+            break;
+        } else {
+            start -= 1;
+        }
+    }
+    start += 1;
+    var end = start;
+    while (end < doc.lineCount) {
+        if (rx.test(doc.lineAt(end).text)) {
+            break;
+        } else {
+            end += 1;
+        }
+    }
+    end -= 1;
+    let startpos = new vscode.Position(start, 0);
+    let endpos = new vscode.Position(end, doc.lineAt(end).text.length);
+    let nextpos = new vscode.Position(end + 1, 0);
+    let code = doc.getText(new vscode.Range(startpos, endpos));
+
+    const params: TextDocumentPositionParams = { 
+        textDocument: vslc.TextDocumentIdentifier.create(ed.document.uri.toString()), 
+        position: startpos
+    }
+
+    const module: string = await g_languageClient.sendRequest('julia/getModuleAt', params);
+
+    await evaluate(ed, new vscode.Range(startpos, endpos), code, module)
+
+    vscode.window.activeTextEditor.selection = new vscode.Selection(nextpos, nextpos)
+    vscode.window.activeTextEditor.revealRange(new vscode.Range(nextpos, nextpos))
+}
+
+async function evaluateBlockOrSelection (shouldMove: boolean = false) {
+    const editor = vscode.window.activeTextEditor
+    const editorId = vslc.TextDocumentIdentifier.create(editor.document.uri.toString());
+
+    for (const selection of editor.selections) {
+        let range: vscode.Range = null
+        let nextBlock: vscode.Position = null
+        const params: TextDocumentPositionParams = { 
+            textDocument: editorId, 
+            position: new vscode.Position(selection.start.line, selection.start.character) 
+        }
+
+        const module: string = await g_languageClient.sendRequest('julia/getModuleAt', params);
+
+        if (selection.isEmpty) {
+            const currentBlock: vscode.Position[] = await g_languageClient.sendRequest('julia/getCurrentBlockRange', params); 
+            range = new vscode.Range(currentBlock[0].line, currentBlock[0].character, currentBlock[1].line, currentBlock[1].character)
+            nextBlock = new vscode.Position(currentBlock[2].line, currentBlock[2].character)
+        } else {
+            range = new vscode.Range(selection.start, selection.end)
+        }
+
+        const text = editor.document.getText(range)
+
+        if (shouldMove && nextBlock && selection.isEmpty && editor.selections.length == 1) {
+            editor.selection = new vscode.Selection(nextBlock, nextBlock)
+            editor.revealRange(new vscode.Range(nextBlock, nextBlock))
+        }
+
+        await evaluate(editor, range, text, module)
+    }
+}
+
+async function evaluate(editor: vscode.TextEditor, range: vscode.Range, text: string, module: string) {
+    await startREPL(true);
+
+    let r = results.addResult(editor, range, {
+        content: ' ⟳ ', 
+        isIcon: false, 
+        hoverContent: '',
+        isError: false
+    })
+
+    let result: any = await g_connection.sendRequest(
```

Comment:
> That's to be expected. We need the LS running to figure out the current code block (and module, once that's in).

hm, the problem I faced often with this branch was like, even after the LS started to run, sometimes the request here gets kinda stuck and after that every inline result won't work (only shows `⟳`.)
This happens if when use some inline execution command just after REPL process started, so I guess there maybe some problem around establishing a connection with REPL.

Reply:
Hm, I haven't seen this issue before. Might be fixed now?

Reply:
Yeah stuck on the initialization seems to have been solve somehow ... 🤔 Let's resolve this for now.